android/src/main/java/com/swmansion/reanimated/nodes/PropsNode.java

```diff
@@ -25,6 +25,28 @@
   private final JavaOnlyMap mPropMap;
   private final ReactStylesDiffMap mDiffMap;
 
+  private void addProp(WritableMap propMap, String key, Object value) {
+    if (value == null) {
+      propMap.putNull(key);
+    } else if (value instanceof Double) {
+      propMap.putDouble(key, (Double) value);
+    } else if (value instanceof Integer) {
+      propMap.putInt(key, (Integer) value);
+    } else if (value instanceof Number) {
+      propMap.putDouble(key, ((Number) value).doubleValue());
+    } else if (value instanceof Boolean) {
+      propMap.putBoolean(key, (Boolean) value);
+    } else if (value instanceof String) {
+      propMap.putString(key, (String) value);
+    } else if (value instanceof WritableArray) {
```

Comment:
Also, it's never another map, no? `Boolean`, `Number` and `String` are enough.

Can we reuse it here https://github.com/kmagiera/react-native-reanimated/pull/311/files#diff-4f72bb0bf303947e60ceb110eb069188R99? It appears to be a similar code.

Reply:
I agree. Can it be `Array` though then? 🤔

Reply:
The logic at https://github.com/kmagiera/react-native-reanimated/pull/311/files#diff-4f72bb0bf303947e60ceb110eb069188R99 seems different, do you think that I can unify both?

Reply:
you are correct!